edb/pgsql/metaschema.py

```diff
@@ -94,6 +111,118 @@ def __init__(self) -> None:
         ),
 
 
+class AlterCurrentDatabaseSetString(dbops.Function):
+    """Alter a configuration parameter of the current database."""
+    text = '''
+    BEGIN
+        EXECUTE 'ALTER DATABASE ' || quote_ident(current_database())
+        || ' SET ' || quote_ident(parameter) || ' = '
+        || coalesce(quote_literal(value), 'DEFAULT');
+        RETURN value;
+    END;
+    '''
+
+    def __init__(self) -> None:
+        super().__init__(
+            name=('edgedb', '_alter_current_database_set'),
+            args=[('parameter', ('text',)), ('value', ('text',))],
+            returns=('text',),
+            volatility='volatile',
+            language='plpgsql',
+            text=self.text,
+        )
+
+
+class AlterCurrentDatabaseSetStringArray(dbops.Function):
+
+    """Alter a configuration parameter of the current database."""
+    text = '''
+    BEGIN
+        EXECUTE 'ALTER DATABASE ' || quote_ident(current_database())
+        || ' SET ' || quote_ident(parameter) || ' = '
+        || coalesce(
+            (SELECT
+                array_to_string(array_agg(quote_literal(q.v)), ',')
+             FROM
+                unnest(value) AS q(v)
+            ),
+            'DEFAULT'
+        );
+        RETURN value;
+    END;
+    '''
+
+    def __init__(self) -> None:
+        super().__init__(
+            name=('edgedb', '_alter_current_database_set'),
+            args=[
+                ('parameter', ('text',)),
+                ('value', ('text[]',)),
+            ],
+            returns=('text[]',),
+            volatility='volatile',
+            language='plpgsql',
+            text=self.text,
+        )
+
+
+class AlterCurrentDatabaseSetNonArray(dbops.Function):
+    """Alter a configuration parameter of the current database."""
+    text = '''
+    BEGIN
+        EXECUTE 'ALTER DATABASE ' || quote_ident(current_database())
+        || ' SET ' || quote_ident(parameter) || ' = '
+        || coalesce(value::text, 'DEFAULT');
+        RETURN value;
+    END;
+    '''
+
+    def __init__(self) -> None:
+        super().__init__(
+            name=('edgedb', '_alter_current_database_set'),
+            args=[
+                ('parameter', ('text',)),
+                ('value', ('anynonarray',)),
+            ],
+            returns=('anynonarray',),
+            volatility='volatile',
+            language='plpgsql',
+            text=self.text,
+        )
+
+
+class AlterCurrentDatabaseSetArray(dbops.Function):
+    """Alter a configuration parameter of the current database."""
+    text = '''
+    BEGIN
+        EXECUTE 'ALTER DATABASE ' || quote_ident(current_database())
+        || ' SET ' || quote_ident(parameter) || ' = '
+        || coalesce(
+            (SELECT
+                array_to_string(array_agg(q.v::text), ',')
```

Comment:
What's the upside of using different encodings for different types, especially unsafe ones, like this, instead of using JSON for all values?

Reply:
1. PostgreSQL settings are always either strings or lists of strings.
2. There's no way to put `ALTER SYSTEM` into a function, so we cannot use JSON encoding for that operation anyway.